util/src/test/scala/org/broadinstitute/dsde/rawls/util/RetrySpec.scala

```diff
@@ -0,0 +1,179 @@
+package org.broadinstitute.dsde.rawls.util
+
+import akka.actor.ActorSystem
+import akka.testkit.TestKit
+import com.typesafe.scalalogging.{LazyLogging, Logger}
+import org.mockito.ArgumentCaptor
+import org.mockito.ArgumentMatchers._
+import org.mockito.Mockito._
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.mock.MockitoSugar
+import org.scalatest.time.{Minutes, Seconds, Span}
+import org.scalatest.{BeforeAndAfterAll, FlatSpecLike, Matchers}
+import org.slf4j.{Logger => Underlying}
+
+import scala.collection.JavaConverters._
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import scala.reflect._
+
+/**
+  * Created by rtitle on 5/16/17.
+  */
+class RetrySpec extends TestKit(ActorSystem("MySpec")) with FlatSpecLike with BeforeAndAfterAll with Matchers with MockitoSugar with ScalaFutures {
+  import system.dispatcher
+
+  implicit override val patienceConfig = PatienceConfig(timeout = scaled(Span(10, Seconds)))
+
+  override def afterAll {
+    TestKit.shutdownActorSystem(system)
+  }
+
+  "Retry" should "retry 3 times by default" in {
+    val testable = new TestRetry(system, setUpMockLogger)
+
+    val result = testable.retry()(() => testable.failure)
+
+    // result should be a failure
+    // invocationCount should be 4 (1 initial run plus 3 retries)
+    // log count should be 4 (3 retries plus 1 failure message)
+    whenReady(result.failed) { ex =>
+      ex shouldBe an [Exception]
+      ex should have message "test exception"
+      testable.invocationCount should equal(4)
+      verify(testable.underlying, times(4)).info(anyString, any[Throwable])
```

Comment:
This is `|+|` all over again :) Could you slap a comment on here explaining what's going on? It took me a while to figure out the meaning, and I _still_ find `verify().info()` bizarre phrasing (shouldn't the thing you're testing against be inside the verify function?).